pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go

```diff
@@ -267,6 +267,33 @@ func getInvolvedObjectTags(involvedObject v1.ObjectReference, taggerInstance tag
 		}
 	}
 
+	var entityID string
+
+	switch involvedObject.Kind {
+	case podKind:
+		entityID = fmt.Sprintf("kubernetes_pod_uid://%s", involvedObject.UID)
+	case deploymentKind:
+		entityID = fmt.Sprintf("deployment://%s/%s", involvedObject.Namespace, involvedObject.Name)
+	default:
+		var apiGroup string
+		apiVersionParts := strings.Split(involvedObject.APIVersion, "/")
+		if len(apiVersionParts) == 2 {
+			apiGroup = apiVersionParts[0]
+		} else {
+			apiGroup = ""
+		}
+		resourceType := strings.ToLower(involvedObject.Kind) + "s"
```

Comment:
Is there a way to get the plural name directly? 

appending an `"s"` can be risky, some resources in kubernetes won't respect this rule, especially CRDs.

Reply:
The involved object doesn't directly expose the resource type. Looked into how we could directly get the resource type, and the only option I've found is to request for all the resources from the Discovery API and sort through them until finding the resource with the matching `Kind` field. We do a similar thing [here](https://github.com/DataDog/datadog-agent/blob/0318c751827c0513dfb1e5f4786076de35f58d5d/comp/core/workloadmeta/collectors/internal/kubeapiserver/utils.go#L156), but I'm guessing we don't want to be doing this every time we go to get the tags. Seeing if we could potentially create a map of `kind --> resource` when the collector starts.

Reply:
Discussed on zoom and agreed to implement a component that facilitates introspection in kubernetes resources, it can use caching, and can be shared by other components. 

For now, we can proceed by appending an "s" and we will create a separate task to work on implementing the shared introspection solution.